tricycle_controller/src/tricycle_controller.cpp

```diff
@@ -0,0 +1,670 @@
+// Copyright 2022 Pixel Robotics.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * Author: Tony Najjar
+ * Author: Sara Al Arab
+ */
+
+#include <memory>
+#include <queue>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
+#include "lifecycle_msgs/msg/state.hpp"
+#include "rclcpp/logging.hpp"
+#include "tf2/LinearMath/Quaternion.h"
+#include "tricycle_controller/tricycle_controller.hpp"
+
+namespace
+{
+constexpr auto DEFAULT_COMMAND_TOPIC = "cmd_vel";
+constexpr auto DEFAULT_COMMAND_UNSTAMPED_TOPIC =
+  "cmd_vel";  // TODO: revert to cmd_vel_unstamped and remap topic
+constexpr auto DEFAULT_COMMAND_OUT_TOPIC = "cmd_vel_out";
+constexpr auto DEFAULT_ACKERMANN_OUT_TOPIC = "ackermann";
+constexpr auto DEFAULT_VALIDATED_ACKERMANN_TOPIC = "validated_ackermann";
+constexpr auto DEFAULT_ODOMETRY_TOPIC = "odom";
+constexpr auto DEFAULT_TRANSFORM_TOPIC = "tf";
```

Comment:
Is there a way to remap these topics from a launch file?

Reply:
yes, check [here](https://github.com/ros-controls/ros2_control_demos/blob/c02942a892dde71f9e2df481b8007ef7f5e173fb/ros2_control_demo_bringup/launch/rrbot.launch.py#L57-L62) for the example

Reply:
In a controller, I always recommend to use local topic , i.e. "~/<topic_name>" so you can remap them from "<controller_name>/<topic_name>" to anything you like

Reply:
Thanks fixed!